src/json/from_json.cr

```diff
@@ -154,13 +154,17 @@ def NamedTuple.new(pull : JSON::PullParser)
     location = pull.location
 
     pull.read_object do |key|
-      case key
-        {% for key, type in T %}
-          when {{key.stringify}}
-            %var{key.id} = {{type}}.new(pull)
-        {% end %}
-      else
-        pull.skip
+      begin
+        case key
+            {% for key, type in T %}
+              when {{key.stringify}}
+                %var{key.id} = {{type}}.new(pull)
+            {% end %}
+        else
+          pull.skip
+        end
+      rescue ex : JSON::ParseException
+        raise ::JSON::ParseException.new("Error parsing attribute #{key} of #{self}: #{ex.message}")
```

Comment:
Is this really expected?

```crystal
require "json"

class Foo
  JSON.mapping({
    foo: Bar
  })
end

class Bar
  JSON.mapping({
    bar: Int32
  })
end

begin
  foo = Foo.from_json(%q{{"foo": {"bar": "baz"}}})
rescue e
  pp e.to_s # => "Error parsing attribute Foo.foo: Error parsing attribute Bar.bar: Expected int but was string at 1:22"
end
```

And so, I suggested to add `attribute_name` property to `JSON::ParseException` to detect such duplication, but I didn't explain this, sorry.

Reply:
I didn't consider that case... I will try to fix that. Thank you again :) 

>And so, I suggested to add attribute_name property to JSON::ParseException to detect such duplication, but I didn't explain this, sorry.

Ah, it makes sense.